libs/langchain/langchain/callbacks/message_log.py

```diff
@@ -0,0 +1,179 @@
+from __future__ import annotations
```

Comment:
This API definitely works as it's information preserving. The usage of jsonpatch is really clever here.